File: vuln-sandbox/php/cwe-89/sql-injection.php
<?php
// CWE-89: SQL Injection via String Concatenation in PHP
// Rule: php-sql-string-concat
// Expected: TruePositive

$conn = new mysqli("localhost", "user", "pass", "db");

// VULNERABLE: User input concatenated into SQL query
$id = $_GET['id'];
$query = "SELECT * FROM users WHERE id = " . $id;
$result = $conn->query($query);

// VULNERABLE: String concatenation with POST data
$username = $_POST['username'];
$sql = "SELECT * FROM accounts WHERE username = '" . $username . "'";
$conn->query($sql);
?>

File: vuln-sandbox/csharp/cwe-89/sql-injection.cs
// CWE-89: SQL Injection via String Concatenation in C#
// Rule: csharp-sql-string-concat
// Expected: TruePositive

using System.Data.SqlClient;

public class UserRepository
{
    private readonly string _connectionString;

    public UserRepository(string connectionString)
    {
        _connectionString = connectionString;
    }

    public object GetUser(string userId)
    {
        using var conn = new SqlConnection(_connectionString);
        conn.Open();

        // VULNERABLE: User input concatenated into SQL command
        var cmd = new SqlCommand("SELECT * FROM Users WHERE Id = " + userId, conn);
        return cmd.ExecuteReader();
    }

    public void UpdateUser(string username, string email)
    {
        using var conn = new SqlConnection(_connectionString);
        conn.Open();

        // VULNERABLE: CommandText built with string concatenation
        var cmd = new SqlCommand("", conn);
        cmd.CommandText = "UPDATE Users SET Email = '" + email + "' WHERE Username = '" + username + "'";
        cmd.ExecuteNonQuery();
    }
}
